### ISSUE TITLE
KeyError - 'PKCS#3_DH_Parameters'

### ISSUE THREAD

**Issue by janderse** [nabla-c0d3/sslyze#77]
When running sslyze v0.11 on Windows 64-bit I get errors for EDH and DHE ciphers:
```
      Undefined - An unexpected error happened: 
                 EDH-RSA-DES-CBC3-SHA                KeyError - 'PKCS#3_DH_Parameters'  
                 DHE-RSA-AES256-SHA                  KeyError - 'PKCS#3_DH_Parameters'  
                 DHE-RSA-AES128-SHA                  KeyError - 'PKCS#3_DH_Parameters'  
```

I did not see this error in previous releases.

I also tried building sslyze v0.11 locally so I could debug.  I used the nassl v0.10 Windows 64-bit pre-built release, since v0.11 is a source-only release.  The error did not occur.  So I am guessing that the bug is actually in nassl v0.11, but I'm not sure.

When the error does not occur I get the following:
```
                 DHE-RSA-AES256-SHA            DH-1024 bits   256 bits      HTTP 200 OK                        
                 DHE-RSA-AES128-SHA            DH-1024 bits   128 bits      HTTP 200 OK                        
                 EDH-RSA-DES-CBC3-SHA          DH-1024 bits   112 bits      HTTP 200 OK                        
```

**Comment by nabla-c0d3** [comment]
Thanks, will look into it

**Comment by LordMike** [comment]
I get the following on a relatively new server. The version is 0.11 (I downloaded a binary release) on a Windows 7 Client.

                 DHE-RSA-AES256-SHA256               KeyError - 'PKCS#3_DH_Parameters'
                 DHE-RSA-AES256-SHA                  KeyError - 'PKCS#3_DH_Parameters'
                 DHE-RSA-AES256-GCM-SHA384           KeyError - 'PKCS#3_DH_Parameters'
                 DHE-RSA-AES128-SHA256               KeyError - 'PKCS#3_DH_Parameters'
                 DHE-RSA-AES128-SHA                  KeyError - 'PKCS#3_DH_Parameters'
                 DHE-RSA-AES128-GCM-SHA256           KeyError - 'PKCS#3_DH_Parameters'

Succesfull algorithms on this server are:

                 ECDHE-RSA-AES256-SHA384       ECDH-256 bits  256 bits                                       
                 ECDHE-RSA-AES256-SHA          ECDH-256 bits  256 bits                                       
                 ECDHE-RSA-AES256-GCM-SHA384   ECDH-256 bits  256 bits                                       
                 ECDHE-RSA-AES128-SHA256       ECDH-256 bits  128 bits                                       
                 ECDHE-RSA-AES128-SHA          ECDH-256 bits  128 bits                                       
                 ECDHE-RSA-AES128-GCM-SHA256   ECDH-256 bits  128 bits

Actually. Looking at it, it seems the suites repeat. So a suite is both accepted and with errors, at the same time?

**Comment by janderse** [comment]
Michael,

DHE and ECDHE are related but different algorithms.  So it isn't
repeating.   sslyze 0.11 has a bug with DHE but not ECDHE algorithms.

**Comment by LordMike** [comment]
I realized my error regarding what I said on them repeating, and edited it shortly after. I must have been sleeping when I compared the two lists. :).

**Comment by jmdesp** [comment]
There's a patch on 0.11 according to which the string "PKCS#3_DH_Parameters" doesn't occur anymore in the source since 'DH_Parameters' is used instead to retrieve the DH parameters :
https://github.com/nabla-c0d3/nassl/commit/9e9d4061356a7e1d370621cac2e5030aa32461f3

Well, I decompressed the content of the windows executable version of 0.11, and if I'm correct, it seems that the patch even if present in github is actually missing from the windows 0.11 release. Which would explain the mismatch with the openssl 1.0.2 release.

**Comment by nabla-c0d3** [comment]
This will be fixed when I release 0.12.
